pkg/providers/instance/instance.go

```diff
@@ -355,8 +355,8 @@ func setVMPropertiesBillingProfile(vmProperties *armcompute.VirtualMachineProper
 	}
 }
 
-// setVMTagsProvisionerName sets "karpenter.sh/provisioner-name" tag
-func setVMTagsProvisionerName(tags map[string]*string, nodeClaim *corev1beta1.NodeClaim) {
+// setVMTagsNodepoolName sets "karpenter.sh/nodepool-name" tag
```

Comment:
It's just `karpenter.sh/nodepool`, should fix the comment + func name.

> Karpenter no longer supports using the `karpenter.sh/provisioner-name` label in NodePool labels and requirements or in application node selectors, affinities, or topologySpreadConstraints. If you were previously using this label to target applications to specific Provisioners, you should update your applications to use the `karpenter.sh/nodepool` label instead before upgrading. If you upgrade without changing these labels, you may begin to see pod scheduling failures for these applications.

Reply:
Also if we want to do a full cleanup, then there are a lot more uses of Provisioner still around: https://github.com/search?q=repo%3AAzure%2Fkarpenter-provider-azure%20provisioner&type=code

Reply:
this was initially just aimed ath the chart and some code, but can do full cleanup now